packages/terminal/src/browser/search/terminal-search-widget.tsx

```diff
@@ -0,0 +1,182 @@
+/********************************************************************************
+ * Copyright (C) 2019 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable, inject, postConstruct } from 'inversify';
+import { ReactWidget } from '@theia/core/lib/browser/widgets/react-widget';
+import * as React from 'react';
+import '../../../src/browser/style/terminal-search.css';
+import { TerminalSearchBox } from '../base/terminal-search-box';
+import { Terminal } from 'xterm';
+import { findNext, findPrevious } from 'xterm/lib/addons/search/search';
+import { ISearchOptions } from 'xterm/lib/addons/search/Interfaces';
+import { Key } from '@theia/core/lib/browser';
+
+export const TERMINAL_SEARCH_WIDGET_FACTORY_ID = 'terminal-search';
+export const TerminalSearchWidgetFactory = Symbol('TerminalSearchWidgetFactory');
+export type TerminalSearchWidgetFactory = (terminal: Terminal) => TerminalSearchWidget;
+
+@injectable()
+export class TerminalSearchWidget extends ReactWidget implements TerminalSearchBox {
+
+    private searchInput: HTMLInputElement | null;
+    private searchBox: HTMLDivElement | null;
+    private searchOptions: ISearchOptions = {};
+
+    @inject(Terminal)
+    protected terminal: Terminal;
+
+    @postConstruct()
+    protected init(): void {
+        this.node.classList.add('theia-search-terminal-widget-parent');
+
+        this.onInputChanged = this.onInputChanged.bind(this);
+        this.onSearchInputFocus = this.onSearchInputFocus.bind(this);
+        this.onSearchInputBlur = this.onSearchInputBlur.bind(this);
+
+        this.onNextButtonClicked = this.onNextButtonClicked.bind(this);
+        this.onPreviousButtonClicked = this.onPreviousButtonClicked.bind(this);
+
+        this.hide = this.hide.bind(this);
+        this.onCaseSensitiveOptionClicked = this.onCaseSensitiveOptionClicked.bind(this);
```

Comment:
please align with coding guidelines: https://github.com/theia-ide/theia/wiki/Coding-Guidelines#no-bind-fn-in-event-handlers

if someone subclass it and override a method it will break, it should be explicit that even handlers are properties, here you implicit convert methods to properties.